yaaredis/pool.py

```diff
@@ -647,3 +648,235 @@ def get_node_by_slot(self, slot):
+
+        channel = options.pop('channel', None)
+
+        if not channel:
+            return await self.get_random_connection()
+
+        slot = self.nodes.keyslot(channel)
+        node = self.get_master_node_by_slot(slot)
+
+        self._checkpid()
+
+        return await self.get_connection_by_node(node)
+
+    def make_connection(self, node):
+        """Creates a new connection"""
+
+        connection = self.connection_class(host=node['host'],
+                                           port=node['port'],
+                                           **self.connection_kwargs)
+
+        # Must store node in the connection to make it eaiser to track
+        connection.node = node
+        return connection
+
+    def release(self, connection):
+        """Releases the connection back to the pool"""
+        self._checkpid()
+        if connection.pid != self.pid:
+            return
+
+        node_name = connection.node['name']
+        self._in_use_connections[node_name].remove(connection)
+        # discard connection with unread response
+        if connection.awaiting_response:
+            connection.disconnect()
+            connection = None
+
+        try:
+            self._available_connections[node_name].put_nowait(connection)
+        except asyncio.QueueFull:
+            # perhaps the pool have been reset() ?
+            pass
+
+    def disconnect(self):
+        """Closes all connections in the pool"""
+        for queue in self._available_connections.values():
+            available_connections = []
+            while True:
+                try:
+                    connection = queue.get_nowait()
+                except asyncio.QueueEmpty:
+                    break
+                connection.disconnect()
+                available_connections.append(connection)
+
+            for connection in available_connections:
+                try:
+                    queue.put_nowait(connection)
+                except asyncio.QueueFull:
+                    pass
+
+        for node_connections in self._in_use_connections.values():
+            for connection in node_connections:
+                connection.disconnect()
+
+    async def get_random_connection(self):
+        """Opens new connection to random redis server"""
+        considered_nodes = list(self._available_connections.keys())
+        random.shuffle(considered_nodes)
+        for node_name in considered_nodes:
+            try:
+                connection = self._available_connections[node_name].get_nowait(
+                )
+                if connection is None:
+                    self._available_connections[node_name].put_nowait(
+                        connection)
+                else:
+                    self._in_use_connections[node_name].add(connection)
+                    return connection
+            except asyncio.QueueEmpty:
+                pass
+
+        # This should only happen if no connection was created yet
```

Comment:
I think this could also occur if all connections are in use across all nodes? Looks like the implementation is fine in that case, but please update the comment to include both cases.